src/app/components/elements/AuthorRecentPosts.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import tt from 'counterpart';
+import { getScotDataAsync } from 'app/utils/steemApi';
+
+import constants from 'app/redux/constants';
+import { LIQUID_TOKEN_UPPERCASE } from 'app/client_config';
+
+const MAX_LIMIT = 10;
+
+const formatDate = date => {
+    const d = new Date(`${date}Z`);
+    const yyyy = d.getFullYear();
+    const mm = String(d.getMonth() + 1).padStart(2, '0');
+    const dd = String(d.getDate()).padStart(2, '0');
+    return `${yyyy}-${mm}-${dd}`;
+};
+
+class AuthorRecentPosts extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            fetching: true,
+            posts: [],
+        };
+    }
+
+    componentDidMount() {
+        this.getDiscussionsByAuthor();
+    }
+
+    async getDiscussionsByAuthor() {
```

Comment:
This violates the pattern of separating view from the actual data powering the view. 

I will likely just add it in to this branch but you should consider placing this in state and triggering a fetch via a Reducer. For example, this will be beneficial if author feeds were powered by similar fetch (though cannot happen at the moment due to resteems not in these feeds. That is in the works also, and will likely need post filtering / query option to make sure this doesn't behave funny)

Reply:
Thank you for answer. 
but I difficult to use a fetch via a Reducer. Could you help me?

Reply:
Ok let me see...  here's how the flow works for 'getContent', which you can mostly mimic

1. dispatch action (in this case REQUEST_DATA):  https://github.com/steemit/condenser/blob/4dc930c9fa07beefcb8013270bb8c670ee1abde4/src/app/components/pages/UserProfile.jsx#L670
1. Action definition:  https://github.com/steemit/condenser/blob/36d87dab1349b24e693e93b6f58a4bedef74825f/src/app/redux/FetchDataSaga.js#L377
1. Add watch for this action to trigger in saga middleware: 
 https://github.com/steemit/condenser/blob/36d87dab1349b24e693e93b6f58a4bedef74825f/src/app/redux/FetchDataSaga.js#L23
1. Function to fetch data https://github.com/steemit/condenser/blob/36d87dab1349b24e693e93b6f58a4bedef74825f/src/app/redux/FetchDataSaga.js#L138  -- note that it uses yield instead of async/await, but syntax is fairly similar. Let me know if you have any troubles here.
1. Request to receive data / store in state:  https://github.com/steemit/condenser/blob/36d87dab1349b24e693e93b6f58a4bedef74825f/src/app/redux/FetchDataSaga.js#L279
1. Definition for action for receive data https://github.com/steemit/condenser/blob/36d87dab1349b24e693e93b6f58a4bedef74825f/src/app/redux/GlobalReducer.js#L490
1. Reducer clause for handling the receive data https://github.com/steemit/condenser/blob/36d87dab1349b24e693e93b6f58a4bedef74825f/src/app/redux/GlobalReducer.js#L267   Here you would define maybe a new state variable like  'recent_user_posts', and map  author -> list of data that you need (looks like titles and permlinks, mainly). I would not recommend 'merging' into existing state, but just replacing the state, so it doesn't accumulate as the user visits many different authors.
1. After this point, state will be updated and will trigger reactivity to show where you are referencing the state data (so in your AuthorRecentPosts you can access it in the mapStateToProps)

I know this sounds like a lot, but I think it's helpful to be consistent, especially if there is a possibility the data can be shared. In any case, let me know if this is something you want to explore or if i should just merge it in, since this does seem specific still (and can be done later I believe. As I also can't think of downsides to current approach at the moment)

Reply:
Thank you so much. 
I will try It.

Reply:
@eonwarped I did it the way you told me. It works well. Thank you so much. You are very kind. I like you.